packages/discovery/src/discovery/provider/DiscoveryProvider.ts

```diff
@@ -56,13 +59,56 @@ export class DiscoveryProvider {
     return Bytes.fromHex(result)
   }
 
-  async getLogs(
+  public async getLogs(
     address: EthereumAddress,
     topics: (string | string[])[],
     fromBlock: number,
     toBlock: number,
   ): Promise<providers.Log[]> {
-    return this.provider.getLogs({
+    if (fromBlock > toBlock) {
+      throw new Error(
+        `fromBlock (${fromBlock}) can't be bigger than toBlock (${toBlock})`,
+      )
+    }
+
+    if (this.getLogsMaxRange === undefined) {
+      return await this.getLogsBatch(address, topics, fromBlock, toBlock)
+    }
+
+    // To support efficient caching, we divide the requested blocks range into
+    // sequential boundaries of `maxRange` size, e.g [0,10k-1], [10k, 20k-1], ...
+    // Otherwise ranges would depend on `fromBlock` and even small change to it
+    // would make the previous cache useless.
+
+    // Let's start with the deployment block number if it's higher than fromBlock
+    const { blockNumber: deploymentBlockNumber } =
+      await this.getDeploymentInfo(address)
+
+    const maxRange = this.getLogsMaxRange
+    const allLogs: providers.Log[][] = []
+
+    let start = Math.max(fromBlock, deploymentBlockNumber)
+    let end
```

Comment:
Why is `end` defined before the scope? Seems like it could very easily be moved to the body of the loop and made const

Reply:
Good catch, it's a leftover from previous implementation. Fixed!